src/parsers/uniprot.py

```diff
@@ -0,0 +1,781 @@
+            association.update(evidence_map[evidence_key])
+        associations.append(clean(association))
+
+    # Catalytic/cofactor
+    for comment in entry.findall("u:comment", NS):
+        comment_type = comment.get("type")
+        if comment_type == "catalytic activity":
+            for reaction in comment.findall("u:reaction", NS):
+                for dbref in reaction.findall("u:dbReference", NS):
+                    db_type = dbref.get("type")
+                    db_id = dbref.get("id")
+                    catalyze_assoc = {
+                        "subject": cdm_id,
+                        "predicate": "catalyzes",
+                        "object": f"{db_type}:{db_id}",
+                        "evidence_type": None,
+                        "supporting_objects": None,
+                        "publications": None,
+                    }
+                    evidence_key = reaction.get("evidence")
+                    if evidence_key and evidence_key in evidence_map:
+                        catalyze_assoc.update(evidence_map[evidence_key])
+                    associations.append(clean(catalyze_assoc))
+
+        elif comment_type == "cofactor":
+            for cofactor in comment.findall("u:cofactor", NS):
+                for dbref in cofactor.findall("u:dbReference", NS):
+                    db_type = dbref.get("type")
+                    db_id = dbref.get("id")
+                    cofactor_assoc = {
+                        "subject": cdm_id,
+                        "predicate": "requires_cofactor",
+                        "object": f"{db_type}:{db_id}",
+                        "evidence_type": None,
+                        "supporting_objects": None,
+                        "publications": None,
+                    }
+                    associations.append(clean(cofactor_assoc))
+    return associations
+
+
+def parse_publications(entry):
+    """
+    Extract all publication references from a UniProt XML <entry>
+    Returns a list of standardized publication IDs (PMID and DOI)
+    """
+    publications = []
+
+    # Iterate through all <reference> blocks in the entry
+    for reference in entry.findall("u:reference", NS):
+        citation = reference.find("u:citation", NS)
+        if citation is not None:
+            # Each <citation> may have multiple <dbReference> elements (e.g., PubMed, DOI)
+            for dbref in citation.findall("u:dbReference", NS):
+                db_type = dbref.get("type")
+                db_id = dbref.get("id")
+                # Standardize format for known publication types
+                if db_type == "PubMed":
+                    publications.append(f"PMID:{db_id}")
+                elif db_type == "DOI":
+                    publications.append(f"DOI:{db_id}")
+
+    return publications
+
+
+def parse_uniprot_entry(entry, cdm_id, datasource_name="UniProt import", prev_created=None):
+    desire_date = datetime.datetime.now(datetime.timezone.utc).isoformat()
```

Comment:
`desire_date` sounds a bit funny! You might want to change it to something like `current_timestamp`. I would also move this so that it gets calculated once, early on in the running of the script.